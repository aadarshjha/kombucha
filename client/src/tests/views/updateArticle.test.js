// Test the Events component
import ReactDOM from "react-dom";
import UpdateArticle from "../../views/login/updateArticle";
import renderer from "react-test-renderer";

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };

// ensures that the events.tsx can itself render.
it("renders newArticle.tsx without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<UpdateArticle />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// checks if the form is rendered
it("renders the newArticle form", () => {
  const div = document.createElement("div");
  ReactDOM.render(<UpdateArticle />, div);
  expect(div.querySelector(".centeredForm")).toBeTruthy();
});

it("renders three form fields", () => {
  const div = document.createElement("div");
  ReactDOM.render(<UpdateArticle />, div);
  expect(div.querySelectorAll(".ant-form-item-label").length).toEqual(2);
});

it("renders the correct form fields", () => {
  const div = document.createElement("div");
  ReactDOM.render(<UpdateArticle />, div);
  expect(
    [...div.querySelectorAll(".ant-form-item-label>label")].map(
      (element) => element.textContent
    )
  ).toEqual(["Title", "Content"]);
});

it("renders the submit button", () => {
  const div = document.createElement("div");
  ReactDOM.render(<UpdateArticle />, div);
  expect(div.querySelectorAll(".ant-btn-primary")).toBeTruthy();
});
