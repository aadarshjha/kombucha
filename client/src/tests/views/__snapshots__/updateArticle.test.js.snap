// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches learn page snapshot 1`] = `
<div
  className="centeredForm"
>
  <form
    className="ant-form ant-form-horizontal"
    id="register"
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={
        Object {
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
        }
      }
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_title"
          title="Title"
        >
          Title
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <input
              className="ant-input"
              id="register_title"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={
        Object {
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
        }
      }
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_content"
          title="Content"
        >
          Content
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <textarea
              className="ant-input"
              id="register_content"
              onChange={[Function]}
              onKeyDown={[Function]}
              rows={4}
              style={Object {}}
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={
        Object {
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
        }
      }
    >
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-16 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Submit Article
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
